### DISCUSSION TITLE
What's the TypeScript type of the handler used in an array event binding?

### DISCUSSION THREAD

**Discussion by jyasskin** [solidjs/solid#1376] [Q&A] [answered]
https://www.solidjs.com/tutorial/bindings_events describes a way to pass an argument to an event handler:

```jsx
const handler = (data, event) => /*...*/

<button onClick={[handler, data]}>Click Me</button>
```

https://github.com/solidjs/solid/blob/c993eb0c6fb7e39772759f1f7605fdb27dbf5863/packages/solid/h/jsx-runtime/src/jsx.d.ts#L43-L52 provides a type for the array as a whole, but I don't see a short way to write the type of the event or the handler. Is there one?

**Comment by genericGenie** [comment, marked as answer]
@jyasskin,
I just bumped into this issue as well. For the type, I specified the array parameter position to use [0] because it has 2 parameters.
I need the event and some optional custom data structure I pass into it. Below is a simplified example.

```
const myHandler: JSX.BoundEventHandler<HTMLDivElement, MouseEvent>[0] = (data, e): void => {
     if (Array.isArray(data))
        console.log(`tagName: ${e.target.tagName} = `, data.join(' '))
     else if (typeof data === 'object' && Object.hasOwn(data, 'myObj'))
        console.log(`tagName: ${e.target.tagName} = `, data.myObj)
}

<div onClick={ [myHandler, ['hello', 'world', 'array']] }>Click Array</div>
<div onClick={ [myHandler, {myObj: 'hello world object'}] }>Click Object</div>

```
